lib/postat.rb

```diff
@@ -6,117 +6,104 @@ class Postat
   attr_reader :client
 
   def initialize
-    @user = POSTAT_CONFIG[:username] || raise('missing :user')
-    @password = POSTAT_CONFIG[:password] || raise('missing :password')
-    @guid = POSTAT_CONFIG[:guid] || raise('missing :guid')
-    @client = Savon.client wsdl: POSTAT_CONFIG[:wsdl], namespace_identifier: :ryl, convert_request_keys_to: :none
+    @client = Savon.client(wsdl: POSTAT_CONFIG[:wsdl],
+                           convert_request_keys_to: :none,
+                           env_namespace: :soapenv,
+                           namespace_identifier: :post,
+                           log: true,
+                           pretty_print_xml: true,
+                           endpoint: POSTAT_CONFIG[:endpoint],
+                           namespaces: {
+                             'xmlns:post' => 'http://post.ondot.at',
+                             'xmlns:arr'  => 'http://schemas.microsoft.com/2003/10/Serialization/Arrays',
+                             'xmlns:core' => 'http://Core.Model',
+                             'xmlns:ser'  => 'http://schemas.microsoft.com/2003/10/Serialization/'
+                           }
+    )
   end
 
   def delete_label(options = {})
-    message = add_common_params
+    message = []
     options[:shipments]&.each do |shipment|
-      message[:Shipments] << { 'CancelRequest.Shipment' => shipment }
+      row = init_message
+      row['post:Number'] = shipment[:reference_number]
+      add_common_params(row)
+      message<< { 'post:CancelShipmentRow' => row }
     end
-    response = client.call(:delete_label, message: { ShippingRequest: message })
-    handle_response response
+    response = client.call(:cancel_shipments, message: { 'post:shipments' => message })
+    handle_response(response, action: :cancel_shipments)
   end
 
+  # NOTE: be careful with the order of attributes to avoid fake missing attribute errors
```

Comment:
seriously??? Go home PostAT, you're drunk!

Reply:
Yes, it was terrible because I spent a lot of time before realizing this situation.